src/db/mod.rs

```diff
@@ -213,7 +194,7 @@ impl Database {
         &self,
         filter: impl Into<Option<Document>>,
         options: impl Into<Option<ListCollectionsOptions>>,
-    ) -> Result<Cursor> {
+    ) -> Result<Cursor<Document>> {
```

Comment:
In Swift, we have this return a `MongoCursor<CollectionSpecification>`, and I think it would be a nice improvement here as well for 2.0. The versioned API will ensure that we won't have to worry about deserialization errors for the foreseeable future.

Reply:
sgtm (though that's probably not surprising since I already agreed to it once...) 

if so, would be nice to use a `DatabaseSpecification` type for `list_databases` too (assuming `listDatabases` is also in API version 1?)

Reply:
Good call, I filed RUST-740 to cover the both of them.